src/streams/WritableStream.ts

```diff
@@ -257,18 +263,23 @@ export default class WritableStream<T> {
 		}
 	}
 
+	/**
+	 * Signals that the producer can no longer write to the stream and it should be immediately moved to an "errored"
+	 * state. Any un-written data that is queued will be discarded.
+	 */
 	abort(reason: any): Promise<void> {
+		// 4.2.4.4-1
 		if (!isWritableStream(this)) {
-			// 4.2.4.4-1
-			return Promise.reject(new TypeError('Must be a WritableStream'));
+			return Promise.reject(
+				new Error('WritableStream method called in context of object that is not a WritableStream instance')
+			);
 		}
 
 		if (this.state === State.Closed) {
 			// 4.2.4.4-2
 			return Promise.resolve();
 		}
-
-		if (this.state === State.Errored) {
+		else if (this.state === State.Errored) {
```

Comment:
Since the previous `if` returns, an else is not needed here.  We have this style of allowing each precondition check to be its own `if` statement.